openstack/manila/templates/type-seeds.yaml

```diff
@@ -15,77 +15,23 @@ spec:
   {{- range $shareTypeName, $shareTypeValues := .Values.share_types }}
-    {{- $enabled := false }}
-    {{- $old_style := false }}
-    {{- if not (kindIs "bool" $shareTypeValues) }}
-      {{- $enabled = $shareTypeValues.enabled }}
-    {{- else }}
-      {{- $old_style = true }}
-      {{- $enabled = $shareTypeValues }}
-    {{- end }}
-    {{- if $enabled }}
-      {{- if $shareTypeName | hasPrefix "hypervisor_storage" }}
+  {{- $shareTypeValues := kindIs "bool" $shareTypeValues | ternary (dict "enabled" $shareTypeValues) $shareTypeValues }}
+  {{- if $shareTypeValues.enabled }}
   - name: {{ $shareTypeName }}
-    is_public: false
-    specs:
-    {{- include "manila_type_seed.specs" . | indent 6 }}
-    extra_specs:
-      share_backend_name: {{ $shareTypeName | quote }}
-    {{- include "manila_type_seed.extra_specs" . | indent 6 }}
-        {{- if not $old_style}}
-          {{- if hasKey $shareTypeValues "extra_specs" }}
-            {{- $shareTypeValues.extra_specs | toYaml | nindent 6 }}
-          {{- else }}
-      provisioning:max_share_size: "32768"
-      provisioning:max_share_extend_size: "32768" # keep this in sync with value "max_asset_sizes" of "- asset_type: 'nfs-shares.*'" in openstack/castellum/templates/configmap.yaml
-          {{- end }}
-        {{- else }}
-      provisioning:max_share_size: "32768"
-      provisioning:max_share_extend_size: "32768" # keep this in sync with value "max_asset_sizes" of "- asset_type: 'nfs-shares.*'" in openstack/castellum/templates/configmap.yaml
-        {{- end }}
-      {{- else if $shareTypeName | eq "standard" }} ## share type "standard"
-  - name: {{ $shareTypeName }}
-    is_public: false
-    description: "Standard"
+    is_public: {{ $shareTypeValues.is_public | default false }}
+    {{- if $shareTypeValues.description }}
+    description: {{ $shareTypeValues.description | quote }}
+    {{- end }}
     specs:
     {{- include "manila_type_seed.specs" . | indent 6 }}
     extra_specs:
-      share_backend_name: {{ $shareTypeName | quote }}
-    {{- include "manila_type_seed.extra_specs" . | indent 6 }}
-        {{- if not $old_style}}
-          {{- if hasKey $shareTypeValues "extra_specs" }}
-            {{- $shareTypeValues.extra_specs | toYaml | nindent 6 }}
-          {{- else }}
-      provisioning:max_share_size: "65536"
-      provisioning:max_share_extend_size: "65536" # keep this in sync with value "max_asset_sizes" of "- asset_type: 'nfs-shares.*'" in openstack/castellum/templates/configmap.yaml
-          {{- end }}
-        {{- else }}
-      provisioning:max_share_size: "65536"
-      provisioning:max_share_extend_size: "65536" # keep this in sync with value "max_asset_sizes" of "- asset_type: 'nfs-shares.*'" in openstack/castellum/templates/configmap.yaml
-        {{- end }}
-      {{- end }}
+    {{ $extraSpecs :=  $.Values.share_type_extra_specs | merge (dict "share_backend_name" $shareTypeName) | merge (default (dict) $shareTypeValues.extra_specs) }}
```

Comment:
this will take 32768 instead of 65536, correct ?

Reply:
first argument takes precedence in `merge` function, so values defined in `.Values.share_types.*.extra_specs` will be rendered